schema.go

```diff
@@ -702,6 +723,7 @@ func makeNodeOf(t reflect.Type, name string, tag []string) Node {
 		list       bool
 		encoded    encoding.Encoding
 		compressed compress.Codec
+		fieldID    *int
```

Comment:
Since id=0 is interpreted as _no field id_, could we use this as well here instead of using a pointer?

Reply:
gentle ping @achille-roussel 

Inside `makeNodeOf` we need to distinguish between  a node has no field id and  a node has `id=0` because we are dealing with user's configuration there is no need for us to interpret anything. 

- no field id : we return the the node as is `(Node).ID()==0`
- `id(0)` : we return a  `fieldIDNode` still `(Node).ID()==0`

Both `(Node).ID()==0` to the public api but internally are two different things.

 Unless I missed something I think `(Node).ID()==0` is the case we explicitly say should be interpreted as  no field id for the node.

Reply:
Having both conditions seem like it will create problems, we need to either define `id=0` as meaning "no field id", or always allow `id=0`.

Since the intent of those ids is to map to protobuf types, I think it's OK to define `id=0` as "no field id" since zero is not a valid field id in protobuf.

Reply:
I see your point now. Indeed protobuf doesn't allow fiel id to be zero.

done 7104886